src/main/java/ro/hiringsystem/service/impl/CandidateUserServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package ro.hiringsystem.service.impl;
+
+
+import ro.hiringsystem.model.CandidateUser;
+import ro.hiringsystem.service.CandidateUserService;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CandidateUserServiceImpl implements CandidateUserService {
+
+    private static Map<UUID, CandidateUser> candidateUserMap = new HashMap<>();
+
+    @Override
+    public Optional<CandidateUser> getById(UUID id) {
+        return candidateUserMap.values().stream()
+                .filter(element -> id.equals(element.getId()))
+                .findAny();
+    }
+
+    @Override
+    public Optional<CandidateUser> getByLastName(Optional<String> lastName) {
+        return candidateUserMap.values().stream()
+                .filter(element -> lastName.equals(element.getLastName()))
+                .findAny();
+    }
+
+    @Override
+    public Map<UUID, CandidateUser> getAllFromMap() {
+        return candidateUserMap;
+    }
+
+    @Override
+    public void addAllFromGivenMap(Map<UUID, CandidateUser> candidateUserMap) {
+        CandidateUserServiceImpl.candidateUserMap.putAll(candidateUserMap);
+    }
+
+    @Override
+    public void addOnlyOne(CandidateUser candidateUser) {
+        candidateUserMap.put(candidateUser.getId(), candidateUser);
+    }
+
+    @Override
+    public void removeElementById(UUID id) {
+        candidateUserMap = candidateUserMap.entrySet().stream()
+                .filter(element -> !id.equals(element.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    @Override
+    public void updateElementById(UUID id, CandidateUser newCandidateUser) {
```

Comment:
This implementation can be moved as a default body for the interface, but since you have the map here, you could just do a put to overwrite the value, or use the compute() method.

The method also currently adds the user if it doesn't exist, since the name of the method is update, you might want make it's return type a boolean and have it return false if the update failed or true if successful.